backend/manager/tools/src/main/java/org/ovirt/engine/core/config/entity/ConfigKeyFactory.java

```diff
@@ -42,7 +42,7 @@ public ConfigKey generateByPropertiesKey(String key) {
         if (StringUtils.isBlank(type)) {
             type = "String";
         }
-        String[] validValues = node.findValuesAsText("validValues").toArray(new String[0]);
+        String[] validValues = node.findValuesAsText("validValues").get(0).split(",");
```

Comment:
What if `findValuesAsText` returns an empty list? The hard-coded call to `get(0)` would probably fail with an exception. 

I think something like this would be more general and safe (not tested):
```
node.findValuesAsText("validValues").stream().map(text -> text.split(",")).flatMap(Stream::of).toArray()
```

Reply:
That was exactly the case, which I tried to take a look when I saw the fix :-)
If there are no validValues defined for the property, then the value is not set and it's not even fetched from db:

```
# engine-config -g FenceStopStatusDelayBetweenRetriesInSec
Picked up JAVA_TOOL_OPTIONS: -Dcom.redhat.fips=false
Index: 0
# engine-config -s FenceStopStatusDelayBetweenRetriesInSec=1
Picked up JAVA_TOOL_OPTIONS: -Dcom.redhat.fips=false
Failed to fetch value of FenceStopStatusDelayBetweenRetriesInSec from DB. Index: 0
# engine-config -g FenceStopStatusDelayBetweenRetriesInSec
Picked up JAVA_TOOL_OPTIONS: -Dcom.redhat.fips=false
Index: 0
```

I will post a fix soon